.github/workflows/codeql-analysis.yml

```diff
@@ -7,34 +7,150 @@ on:
     # The branches below must be a subset of the branches above
     branches: [master]
   schedule:
-    - cron: '0 1 * * 4'
+    # At 01:37 every Thursday. Details in https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule
+    - cron: '37 1 * * 4'
 
 jobs:
+  create-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get languages from repo
+        id: set-matrix
+        uses: advanced-security/set-codeql-language-matrix@v1
+        with:
+          access-token: ${{ github.token }}
+          endpoint: ${{ github.event.repository.languages_url }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.languages }}
+
   analyze:
     name: Analyze
-    if: github.repository == 'photoview/photoview'
-    runs-on: ubuntu-20.04
-
-    # strategy:
-    #   fail-fast: false
-    #   matrix:
-        # Override automatic language detection by changing the below list
-        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
-        # language: ['go', 'javascript']
-        # Learn more...
-        # https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#overriding-automatic-language-detection
-
+    needs: create-matrix
+    if: ${{ needs.create-matrix.outputs.matrix != '[]' && github.repository == 'photoview/photoview' }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ${{ fromJSON(needs.create-matrix.outputs.matrix) }}
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+
+      - name: Install GO build dependencies
+        if: ${{ matrix.language == 'go' }}
+        run: |
+          sudo apt-get update
```

Comment:
Better to call `./scripts/install_build_dependencies.sh`.

Reply:
I tried a lot of things (check the history of recent commits here), but cannot install the dependencies: https://github.com/photoview/photoview/actions/runs/10633902870/job/29480008655?pr=1028
Do you have some ideas on how to solve this issue?

Reply:
Checking.

Reply:
I think the problem is because we build API with `golang:1.22-bookworm` and add testing repos but here it's debian-testing only. Some packages are removed in the latest `debian-testing`.

You can find how API sets env [here](https://github.com/photoview/photoview/blob/master/Dockerfile#L51). One possible way is that using `container: go:1.22-bookwrom` and copying [these code](https://github.com/photoview/photoview/blob/master/Dockerfile#L51) to build.

I'm checking if there is a way to build api from `debian-testing` and install golang from `debian-testing`.

Reply:
> the problem is because we build API with `golang:1.22-bookworm` and add testing repos but here it's debian-testing only.

I tried both base images with the same result(

> You can find how API sets env [here](https://github.com/photoview/photoview/blob/master/Dockerfile#L51)

Yes, I used that flow as the source for implementing this one

> One possible way is that using `container: go:1.22-bookwrom` and copying [these code](https://github.com/photoview/photoview/blob/master/Dockerfile#L51) to build.

Tried that in previous commits with the same result(

I'll try removing that huge folder (to check the assumption of no space left) and if it won't help, try to build the image target 'api' and use it for this job

Reply:
@googollee I thought that there is something wrong with my Bash script to install deps into the image, but I tried to build locally the PROD image from the Dockerfile and got the same issue, as failed all recent runs (since y-day's evening): https://github.com/photoview/photoview/actions/runs/10643539201/job/29507209980?pr=1028
With this, it looks like the `debian:testing` image is actually testing for Debian team and they allow it to fail from time to time. For now, all other workflows complete successfully only because of cached steps.
Please double-check and confirm from your side.
If that is true, we cannot rely on it and have to move to another base image before the release (and probably before merging something else to the `master`)

Reply:
Agreed. How about alpine? I was able to build with alpine before. Alpine releases stable every half year, and we could expect upgrading the alpine every half year. 

I don't know if jellyfin- ffmpeg works well with alpine, since it doesn't provide a build with musl.

Reply:
Sure, that was our 2nd option, so let's try.